onyo/lib/command_utils.py

```diff
@@ -72,72 +68,47 @@ def sanitize_keys(k: Optional[list[str]],
     return deduplicate(k) if k else defaults
 
 
-def set_filters(
-        filters: list[str], repo: OnyoRepo, rich: bool = False) -> list[Filter]:
-    """Create filters and check if there are no duplicate filter keys"""
-    # Note: This is part of the get command
-
-    init_filters = []
-    try:
-        init_filters = [Filter(f) for f in filters]
-    except OnyoInvalidFilterError as exc:
-        if rich:
-            console = Console(stderr=True)
-            console.print(f'[red]FAILED[/red] {exc}')
-        else:
-            ui.print(exc, file=sys.stderr)
-        # TODO: This raise replaces a sys.exit; Ultimately error messages above should be integrated in exception and
-        #       rendering/printing handled upstairs.
-        raise
-
-    # ensure there are no duplicate filter keys
-    duplicates = [
-        x for x, i in Counter([f.key for f in init_filters]).items() if i > 1]
-    if duplicates:
-        if rich:
-            console = Console(stderr=True)
-            console.print(
-                f'[red]FAILED[/red] Duplicate filter keys: {duplicates}')
-        else:
-            ui.print(f'Duplicate filter keys: {duplicates}', file=sys.stderr)
-        # TODO: This raise replaces a sys.exit; Ultimately error messages above should be integrated in exception and
-        #       rendering/printing handled upstairs.
-        raise ValueError
-    return init_filters
+def fill_unset(assets: Generator[dict, None, None] | filter, keys: list) -> Generator:
+    """Fill values for missing `keys` in `assets` with `UNSET_VALUE`.
 
+    Helper for the onyo-get command.
 
-def fill_unset(
-        assets: Generator[tuple[Path, dict[str, str]], None, None],
-        keys: list, unset: str = UNSET_VALUE) -> Generator:
+    Parameters
+    ----------
+    assets: Generator
+      Asset dictionaries to fill.
+    keys: list of str
+      Keys for which to set `UNSET_VALUE` if not present in an asset.
     """
-    If a key is not present for an asset, define it as `unset`.
-    """
-    unset_keys = {key: unset for key in keys}
-    for asset, data in assets:
-        yield asset, unset_keys | data
+    for asset in assets:
+        yield {k: UNSET_VALUE for k in keys} | asset
 
 
 def natural_sort(
-        assets: list[tuple[Path, dict[str, str]]],
+        assets: list[dict],
         keys: Optional[list] = None, reverse: bool = False) -> list:
```

Comment:
It's not stated, because it shouldn't matter. It sorts a list of things. Anything can be in there as long as it can be converted to a string in this implementation.